src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,513 @@
+sidebar:
+  order: 138
+  label: IdentityServer4 → v7.2
+---
+
+import { Badge } from "@astrojs/starlight/components";
+import { Code } from "astro/components";
+import { Steps } from "@astrojs/starlight/components";
+import { Tabs, TabItem } from "@astrojs/starlight/components";
+
+This upgrade guide covers upgrading from IdentityServer4 to Duende IdentityServer v7.2.
+IdentityServer4 reached its end of life (EOL) on December 13, 2022. It is strongly advised to migrate to Duende IdentityServer.
+
+Depending on your current version of IdentityServer4, different steps may be required.
+You can determine the version of IdentityServer4 by running the `dotnet list` command at the root of your project, or using NuGet tooling in Visual Studio or JetBrains Rider.
+
+<Tabs>
+  <TabItem label="Windows">
+    <Code
+      code={'dotnet list package | sls "IdentityServer4"'}
+      lang="bash"
+      title="Terminal"
+    />
+  </TabItem>
+  <TabItem label="macOS / Linux">
+    <Code
+      code={"dotnet list package | grep IdentityServer4"}
+      lang="bash"
+      title="Terminal"
+    />
+  </TabItem>
+</Tabs>
+
+This command will print a list of packages you are using in your solution, along with their version.
+
+```bash title="Output"
+   > IdentityServer4                            3.1.4       3.1.4
+   > IdentityServer4.EntityFramework            3.1.4       3.1.4
+   > IdentityServer4.AccessTokenValidation      3.0.1       3.0.1
+   > IdentityServer4.AccessTokenValidation      2.6.0       2.6.0
+```
+
+Depending on the package version shown, your next steps will be different:
+
+- If you are on IdentityServer v3.x, we recommend first [upgrading to IdentityServer4 v4.x](#identityserver4-v3x-to-identityserver-v4x), and then to [Duende IdentityServer](#identityserver4-v4x-to-duende-identityserver). The configuration object model changed between the two major versions of IdentityServer4, and we recommend upgrading step-by-step.
+- If you are on IdentityServer v4.x, you can immediately [upgrade to Duende IdentityServer](#identityserver4-v4x-to-duende-identityserver).
+
+## IdentityServer4 v3.x to IdentityServer v4.x
+
+The most straightforward upgrade path is to first update to the latest version of IdentityServer4 v4, and then continue to the latest Duende IdentityServer.
+
+Between IdentityServer4 v3.x and v4.x, the configuration object model was updated:
+
+- The relation between `ApiResources` and `ApiScopes` was changed from parent-child to many-to-many.
+- A number of configuration types were renamed:
+  - `ApiProperties` to `ApiResourceProperties`
+  - `ApiSecrets` to `ApiResourceSecrets`
+  - `IdentityClaims` to `IdentityResourceClaims`
+  - `IdentityProperties` to `IdentityResourceProperties`
+  - `ApiScopes` to `ApiResourceScopes`
+
+IdentityServer4 projects that use the `IdentityServer4.EntityFramework` package or implement their own stores will need to update their code and/or database to reflect these changes.
```

Comment:
Make it more clear that we have a custom migration script. Don't run the generated DB migration for our EF store as it will loose data.

I see we have it further down, maybe just a reference here that there are details coming, to prevent cowboys from just applying the generated migration.